src/wasm-linker.h

```diff
@@ -32,6 +32,9 @@ namespace wasm {
 
 class S2WasmBuilder;
 
+// Name of the dummy function to prevent erroneous nullptr comparisons.
+static constexpr const char* dummyFunction = "__wasm_nullptr";
```

Comment:
More middle-of-the-road, lets strip `dummyFunction` before calling into wasm-emscripten (https://github.com/WebAssembly/binaryen/commit/1c5fdb97b7b90327c87f6434d8163f19ec9fb423). That way, the linker can exclude what it wants to exclude, and the i64-exclusion can still be embedding-specific knowledge.